java-chassis-reference/en_US/general-development/file-upload.md

```diff
@@ -58,25 +58,25 @@ public String fileUpload(@RequestPart(name = "file1") MultipartFile file1, @Requ
 
 * org.springframework.core.io.Resource
 
-使用InputStream时，因为是流的方式，此时没有客户端文件名的概念，所以producer获取客户端文件名会得到null
+When using InputStream, because it is a stream, there is no concept of client file name at this time, so the producer will get the client file name will get null.
 
-如果既要使用内存数据，又想让producer可以获取客户端文件名，可以使用resource类型，继承org.springframework.core.io.ByteArrayResource，且override getFilename即可。
+If you want to use both memory data and the producer to get the client file name, you can use the resource type, inherit org.springframework.core.io.ByteArrayResource, and override getFilename.
```

Comment:
If you want to use both memory data and the producer to get the client file name ->
If you want to use memory data and the producer also can get the client file name